test/unit/app/controllers/permissions/permissions-middleware-test.js

```diff
@@ -0,0 +1,752 @@
+
+      grantPermissions(
+        permController, ORIGINS.c,
+        PERMS.finalizedRequests.eth_accounts(ACCOUNT_ARRAYS.c)
+      )
+
+      const req = RPC_REQUESTS.eth_requestAccounts(ORIGINS.c)
+      const res = {}
+
+      await assert.doesNotReject(
+        cMiddleware(req, res),
+        'should not reject'
+      )
+
+      assert.ok(
+        res.result && !res.error,
+        'response should have result and no error'
+      )
+      assert.deepEqual(
+        res.result, ACCOUNT_ARRAYS.c,
+        'response should have correct result'
+      )
+    })
+  })
+
+  describe('wallet_sendDomainMetadata', function () {
+
+    let permController
+
+    beforeEach(function () {
+      permController = initPermController()
+    })
+
+    it('records domain metadata', async function () {
+
+      const name = 'BAZ'
+
+      const cMiddleware = getPermissionsMiddleware(permController, ORIGINS.c)
+
+      const req = RPC_REQUESTS.wallet_sendDomainMetadata(ORIGINS.c, name)
+      const res = {}
+
+      await assert.doesNotReject(
+        cMiddleware(req, res),
+        'should not reject'
+      )
+
+      assert.ok(res.result, 'result should be true')
+
+      const metadataStore = permController.store.getState()[METADATA_STORE_KEY]
+
+      assert.deepEqual(
+        metadataStore,
+        { [ORIGINS.c]: { name, extensionId: undefined } },
+        'metadata should have been added to store'
+      )
+    })
+
+    it('records domain metadata and preserves extensionId', async function () {
+
+      const extensionId = 'fooExtension'
+
+      const name = 'BAZ'
+
+      const cMiddleware = getPermissionsMiddleware(permController, ORIGINS.c, extensionId)
+
+      const req = RPC_REQUESTS.wallet_sendDomainMetadata(ORIGINS.c, name)
+      const res = {}
+
+      await assert.doesNotReject(
+        cMiddleware(req, res),
+        'should not reject'
+      )
+
+      assert.ok(res.result, 'result should be true')
+
+      const metadataStore = permController.store.getState()[METADATA_STORE_KEY]
+
+      assert.deepEqual(
+        metadataStore,
+        { [ORIGINS.c]: { name, extensionId } },
+        'metadata should have been added to store'
+      )
+    })
+
+    it('should have record domain metadata if no name', async function () {
+
+      const name = null
+
+      const cMiddleware = getPermissionsMiddleware(permController, ORIGINS.c)
+
+      const req = RPC_REQUESTS.wallet_sendDomainMetadata(ORIGINS.c, name)
+      const res = {}
+
+      await assert.doesNotReject(
+        cMiddleware(req, res),
+        'should not reject'
+      )
+
+      assert.ok(res.result, 'result should be true')
+
+      const metadataStore = permController.store.getState()[METADATA_STORE_KEY]
+
+      assert.deepEqual(
+        metadataStore, {},
+        'metadata should not have been added to store'
+      )
+    })
+
+    it('should have record domain metadata if no metadata', async function () {
```

Comment:
Same as comment above; I'm guessing this is what was meant by this test description - I find it confusing as-written.
```suggestion
    it('should not record domain metadata if no metadata', async function () {
```

Reply:
Fixed in 991ca6e